slides/linear_model.md

```diff
@@ -0,0 +1,260 @@
+
+Left: As we can see, our linear model captures and amplifies the noise in the data. It displays a lot of *variance*.
+
+Right: Ridge estimator regularizes the coefficients by shrinking lightly them to zero
+
+Ridge displays much less variance. However it systematically under-estimates the coefficient. It displays a *biased* behavior.
+
+This is a typical example of bias/variance tradeof: non-regularized estimator are not biased, but they can display a lot of variance. Highly-regularized models have little variance, but high bias. This bias is not necessarily a bad thing: what matters is choosing the tradeoff between bias and variance that leads to the best prediction performance. For a specific dataset there is a sweet spot corresponding to the highest complexity that the data can support, depending on the amount of noise and of observations available.
+
+---
+# 2 points example
+
+
+.pull-left.shift-left[<img src="../figures/lin_reg_2_points_no_penalty_grey.svg" width="110%">]
+.pull-right[<img src="../figures/lin_reg_2_points_ridge_grey.svg" width="110%">]
+
+.pull-left.shift-left[     Linear regression]
+.pull-right[         Ridge]
+.pull-left.shift-left[     No bias, high variance]
+.pull-right[         low bias, low variance]
+???
+
+
+---
+# Regularization in logistic regression
+The parameter C of logistic regression controls the "complexity" of the model, and in practice whether the model focuses on data close to the boundary.
+
+.shift-left.pull-left[<img src="../figures/logistic_2D_C0.001.svg" width="100%">]
+.pull-right[<img src="../figures/logistic_2D_C1.svg" width="100%">]
+.pull-left.shift-left[     Small C]
+.pull-right[         Large C]
+???
+For large value of C, the model put more emphasize on the frontier's point. 
+In contrary, for low value of C, the model is considering all the points.
+The choice of C depends of the dataset.
+
+---
+# Logistic regression for multiclass
+
+Logistic regression could adapt even if **y** contains multiple classes. There are several options:
+
+.shift-left.pull-left[<img src="../figures/multinomial.svg" width="100%">]
+.pull-right[
+* Multinomial 
+* One vs One
+* One vs Rest
+]
+???
+Mutlinomial logistic regression is a natural extension of logistic regression.
+Otherwise, we stil can run One vs Rest approach.
+
+---
+# Linear separability
+
+
+.shift-left.pull-left[<img src="../figures/lin_separable.svg" width="100%">]
+.pull-right[<img src="../figures/lin_not_separable.svg" width="100%">]
+
+.pull-left.shift-left[Linearly separable]
+.pull-right[     linearly *Not* separable]
+
+???
+Linear models work as long as your data could be linearly separable.
+Otherwise, either we could do features augmentation, or we could choose a more complex model.
+
+---
+.center[
+# Take home messages 
+]
+
+* Linear model are good baselines for:
+ - regression: linear regression + penalty = Ridge
+ - classification: logistic regression
+
+* very fast to train
+* Better when *p* > *n*
```

Comment:
I think we should use number of features and number of samples instead of `p` and `n`